src/_pytest/assertion/rewrite.py

```diff
@@ -703,24 +703,15 @@ def run(self, mod: ast.Module) -> None:
             return
         pos = 0
         for item in mod.body:
-            if (
-                expect_docstring
-                and isinstance(item, ast.Expr)
-                and isinstance(item.value, ast.Constant)
-                and isinstance(item.value.value, str)
-            ):
-                doc = item.value.value
-                if self.is_rewrite_disabled(doc):
-                    return
-                expect_docstring = False
-            elif (
-                isinstance(item, ast.ImportFrom)
-                and item.level == 0
-                and item.module == "__future__"
-            ):
-                pass
-            else:
-                break
+            match item:
+                case ast.Expr(value=ast.Constant(value=str(doc))) if expect_docstring:
```

Comment:
```suggestion
                case ast.Expr(value=ast.Constant(value=str() as doc)) if expect_docstring:
```
This will be faster even if it doesn't look as nice. `str(doc)` will always do the `__match_args__` lookup which will be slow.